spampd.pl

```diff
@@ -543,25 +552,35 @@ sub init {
   # If debug output requested, do it now and exit.
   $self->show_debug($spd_p->{show_dbg}, {$self->options_map()}, \@startup_args) && exit(0) if $spd_p->{show_dbg};
 
-  # Create and set up SpamAssassin object. This replaces our SpamPD->{assassin} property with the actual object instance.
-  $sa_p = Mail::SpamAssassin->new($sa_p);
-
-  $spd_p->{sa_awl} and eval {
-    require Mail::SpamAssassin::DBBasedAddrList;
-    # create a factory for the persistent address list
-    my $addrlistfactory = Mail::SpamAssassin::DBBasedAddrList->new();
-    $sa_p->set_persistent_address_list_factory($addrlistfactory);
-  };
+  my $sa_rules_ver;
+  if ($spd_p->{sa_client}) {
+    $sa_c = Mail::SpamAssassin::Client->new($sa_c);
+    $self->{assassinc} = $sa_c;
+    $self->inf("Pinging sa daemon");
+    if ($sa_c->ping()){
```

Comment:
What happens if it doesn't respond?  Is the whole startup useless (or will fail?) or will it connect to spamd later?

Reply:
as per https://github.com/apache/spamassassin/blob/e088d9e28c5342cb9b31daa2b0516425eaaa1d68/lib/Mail/SpamAssassin/Client.pm#L590 client create new connection for each request.

Reply:
@catap yes, you are correct if spamd is doesn't running yet it will slow down. Removing this change